cmd/status.go

```diff
@@ -0,0 +1,281 @@
+	"github.com/go-gota/gota/series"
+
+	"github.com/spf13/cobra"
+)
+
+const statusDesc = `
+Get status of a workflow.
+
+The ` + "``status``" + ` command allow to retrieve status of a workflow. The status
+can be created, queued, running, failed, etc. You can increase verbosity or
+filter retrieved information by passing appropriate command-line options.
+
+Examples:
+
+  $ reana-client status -w myanalysis.42
+
+  $ reana-client status -w myanalysis.42 -v --json
+`
+
+const statusFormatFlagDesc = `Format output by displaying only certain columns.
+E.g. --format name,status.
+`
+
+type statusOptions struct {
+	token           string
+	serverURL       string
+	workflow        string
+	formatFilters   []string
+	jsonOutput      bool
+	verbose         bool
+	includeDuration bool
+}
+
+// newStatusCmd creates a command to get status of a workflow.
+func newStatusCmd() *cobra.Command {
+	o := &statusOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Get status of a workflow.",
+		Long:  statusDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if o.token == "" {
+				o.token = os.Getenv("REANA_ACCESS_TOKEN")
+			}
+			o.serverURL = os.Getenv("REANA_SERVER_URL")
+			if o.workflow == "" {
+				o.workflow = os.Getenv("REANA_WORKON")
+			}
+
+			if err := validation.ValidateAccessToken(o.token); err != nil {
+				return err
+			}
+			if err := validation.ValidateServerURL(o.serverURL); err != nil {
+				return err
+			}
+			if err := validation.ValidateWorkflow(o.workflow); err != nil {
+				return err
+			}
+			if err := o.run(cmd); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&o.token, "access-token", "t", "", "Access token of the current user.")
+	f.StringVarP(
+		&o.workflow,
+		"workflow",
+		"w", "",
+		"Name or UUID of the workflow. Overrides value of REANA_WORKON environment variable.",
+	)
+	f.StringSliceVar(&o.formatFilters, "format", []string{}, statusFormatFlagDesc)
+	f.BoolVar(&o.jsonOutput, "json", false, "Get output in JSON format.")
+	f.BoolVarP(&o.verbose, "verbose", "v", false, "Set status information verbosity.")
+	f.BoolVar(
+		&o.includeDuration,
+		"include-duration",
+		false,
+		`Include the duration of the workflows in seconds.
+In case a workflow is in progress, its duration as of now will be shown.`,
+	)
+
+	return cmd
+}
+
+func (o *statusOptions) run(cmd *cobra.Command) error {
+	statusParams := operations.NewGetWorkflowStatusParams()
+	statusParams.SetAccessToken(&o.token)
+	statusParams.SetWorkflowIDOrName(o.workflow)
+
+	api, err := client.ApiClient()
+	if err != nil {
+		return err
+	}
+	statusResp, err := api.Operations.GetWorkflowStatus(statusParams)
+	if err != nil {
+		return err
+	}
+	payload := statusResp.Payload
+
+	header := buildStatusHeader(
+		o.verbose,
+		o.includeDuration,
+		payload.Progress,
+		payload.Status,
+	)
+	parsedFormatFilters := utils.ParseFormatParameters(o.formatFilters, false)
```

Comment:
While doing this, I noticed the Python client doesn't respect order in the format filter. Should we open an issue for that?

Reply:
Yes please!